scripts/apps/desks/directives/DesksReactDropdown.jsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * @ngdoc directive
+ * @module superdesk.apps.desks
+ * @name DesksReactDropdown
+ *
+ * @requires React
+ * @requires item
+ * @requires className
+ * @requires desks
+ * @requires noDesksLabel
+ *
+ *
+ * @description Creates dropdown react element with list of available desks
+ */
+import React from 'react';
+
+DesksReactDropdown.$inject = ['item', 'className', 'desks', 'noDesksLabel'];
+export function DesksReactDropdown(item, className, desks, noDesksLabel) {
+    /** Dropdown component */
+    class MarkDesksDropdown extends React.Component {
+        render() {
+            let items = [];
+
+            if (desks.desks._items.length) {
+                desks.desks._items.forEach((d) => {
+                    items.push(<MarkedDeskItem key={d._id} desk={d} item={item}/>);
+                });
+            } else {
+                items.push(<li><button disabled="true">{noDesksLabel}</button></li>);
+            }
+
+            return (
+                <ul className={this.props.className}>
+                    {items}
+                </ul>
+            );
+        }
+    }
+
+    /** Button component for marking items for desks */
+    class MarkBtn extends React.Component {
```

Comment:
Each component should be in its own file. There is a pending PR which will be merged soon that will disallow this via linting anyway, so might as well do it now.